staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -154,7 +178,17 @@ func (f *FieldManager) Apply(liveObj, appliedObj runtime.Object, manager string,
 
 	internal.RemoveObjectManagedFields(liveObj)
 
-	if object, managed, err = f.fieldManager.Apply(liveObj, appliedObj, managed, manager, force); err != nil {
+	if userAgent == "kubectl" && f.clientSideApplyUpgrader != nil {
+		// Upgrade the client-side apply annotation
+		// only if the apply is from kubectl
+		object, managed, err = f.upgradeClientSideApply(liveObj, appliedObj, managed, manager, force)
```

Comment:
This name is a little misleading. `ApplyUsingLastAppliedAnnotation`? Or something that makes it clear that this also applies.

Reply:
That's better, renamed